unblob/models.py

```diff
@@ -63,6 +65,13 @@ def size(self) -> int:
     def range_hex(self) -> str:
         return f"0x{self.start_offset:x}-0x{self.end_offset:x}"
 
+    @property
+    def is_whole_file(self):
+        assert self.host_file
+        return (
+            self.start_offset == 0 and self.end_offset == self.host_file.stat().st_size
```

Comment:
can't we use the `size()` call to get the file size? Our `File` has a size method which also works for the `from_bytes()` setup as well.

Also it might make sense to cache this, instead of calling stat() multiple times

Reply:
`stat` would have been called at most twice for ELF files, and only once for all the rest, so it was not a really expensive thing, but changing from `Path` to `File` made it unnecessary.

----

One drawback of using `File`, is that with `Path` the option to process chunks in parallel was there, with `File` it is not: `File` objects will not travel through process boundaries.

Reply:
File is used now.